File: Go/ExampleTests/types_test.go
package exampletests

import (
	"testing"
	"exampletests/common"
)

func TestTypes_DataType_Dollar(t *testing.T) {
	glue := NewTypesGlue()
	objectList1 := []common.ValidValuesString{
		common.NewValidValuesStringFromSlice([]string{"0", "true", ""}),
		common.NewValidValuesStringFromSlice([]string{"0.01", "true", ""}),
		common.NewValidValuesStringFromSlice([]string{"-1", "false", "Negative not allowed"}),
		common.NewValidValuesStringFromSlice([]string{"0.001", "false", "Only 2 decimal digits"}),
	}
	glue.ExamplesDataTypeDollar(t, objectList1)
}

func TestTypes_DataType_SimpleText(t *testing.T) {
	glue := NewTypesGlue()
	objectList2 := []common.ValidValuesString{
		common.NewValidValuesStringFromSlice([]string{"abc", "y", ""}),
		common.NewValidValuesStringFromSlice([]string{"ab.", "y", "period okay"}),
		common.NewValidValuesStringFromSlice([]string{"1234567890", "y", "digits"}),
		common.NewValidValuesStringFromSlice([]string{"@", "n", ""}),
		common.NewValidValuesStringFromSlice([]string{"-a-b", "y", "hyphens"}),
	}
	glue.ExamplesDataTypeSimpleText(t, objectList2)
}

File: Java/SpectableJavaExamplesTest/src/test/java/spectable/Types_Test.java
package spectable.tests;

import java.util.List;
import java.util.ArrayList;
import spectable.common.*;
import spectable.Types_glue;
import production.*;
import org.junit.jupiter.api.Test;

public class Types_Test {

    // -------------------------
    // DataType Tests
    // -------------------------
    @Test
    public void DataType_Dollar() {
        Types_glue glue = new Types_glue();
        List<ValidValuesString> objectList1 = new ArrayList<>();
        objectList1.add(new ValidValuesString("0", "true", ""));
        objectList1.add(new ValidValuesString("0.01", "true", ""));
        objectList1.add(new ValidValuesString("-1", "false", "Negative not allowed"));
        objectList1.add(new ValidValuesString("0.001", "false", "Only 2 decimal digits"));
        glue.Examples_DataType_Dollar(objectList1);
    }

    @Test
    public void DataType_SimpleText() {
        Types_glue glue = new Types_glue();
        List<ValidValuesString> objectList2 = new ArrayList<>();
        objectList2.add(new ValidValuesString("abc", "y", ""));
        objectList2.add(new ValidValuesString("ab.", "y", "period okay"));
        objectList2.add(new ValidValuesString("1234567890", "y", "digits"));
        objectList2.add(new ValidValuesString("@", "n", ""));
        objectList2.add(new ValidValuesString("-a-b", "y", "hyphens"));
        glue.Examples_DataType_SimpleText(objectList2);
    }

}
